src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,46 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.handleAdParams_(eventData);
+    super.onInit(eventData);
+  }
+
+  /**
+   * Checks validity of ad parameters and rejects with proper message if invalid.
+   * @param eventData an object that contains information details for a particular event
+   *   such as event type, unique Ids, creativeData and environmentData.
+   * @private 
+  */ 
+  handleAdParams_(eventData) {
```

Comment:
you could even name this function more specifically

validateAdParams_
or
validateAndParseAdParams_

Reply:
changed